distage/distage-model/src/main/scala/com/github/pshirshov/izumi/distage/model/definition/dsl/AbstractBindingDefDSL.scala

```diff
@@ -40,11 +42,60 @@ trait AbstractBindingDefDSL[BindDSL[_], MultipleDSL[_], SetDSL[_]] {
     _bindDSL[T](ref)
   }
 
-  final protected def bindMany[T: Tag](implicit pos: CodePositionMaterializer): MultipleDSL[T] = {
-    val ref = registered(new MultipleRef(Bindings.binding[T]))
-    _multipleDSL[T](ref)
+  final protected def bind[T: Tag](implicit pos: CodePositionMaterializer): MultipleDSL[T] = {
+   bindImpl[T](ImplDef.TypeImpl(SafeType.get[T]))
+  }
+
+  final protected def bind[T: Tag](instance: T)(implicit pos: CodePositionMaterializer): MultipleDSL[T] = {
+    bindImpl[T](ImplDef.InstanceImpl(SafeType.get[T], instance))
+  }
+
+  final protected def bind[T: Tag](function: ProviderMagnet[T])(implicit pos: CodePositionMaterializer): MultipleDSL[T] = {
+    bindImpl[T](ImplDef.ProviderImpl(SafeType.get[T], function.get))
+  }
+
+  final protected def bindReference[T: Tag](implicit pos: CodePositionMaterializer): MultipleDSL[T] = {
```

Comment:
I think you can omit `bindReference` since that would create a double reference which is not exactly a great idea.